### PR TITLE
fix issue #2278

### PR DESCRIPTION
**Pull request by watashiSHUN**
Kudu.FunctionalTests.SiteExtensions.SiteExtensionApiFacts.SiteExtensionV2AndV3FeedTests
Kudu.FunctionalTests.SiteExtensions.SiteExtensionApiFacts.SiteExtensionGetArmTest
Kudu.FunctionalTests.SiteExtensions.SiteExtensionApiFacts.SiteExtensionBasicTests
....

which all calls [GetRemoteExtensions](https://github.com/projectkudu/kudu/blob/4593ba51254eaca21fe7f2274718a995239980c2/Kudu.FunctionalTests/SiteExtensions/SiteExtensionApiFacts.cs#L96) should no longer appear in teamcity test errors

### CODE REVIEW

File: Kudu.Core/SiteExtensions/FeedExtensions.cs [hunk 1]
```diff
@@ -57,6 +57,23 @@ public static class FeedExtensions
 
             return latestPackage;
         }
+        
+        // TODO refactor last function, after kudu private test all passes
```
Comment (davidebbo) [inline Kudu.Core/SiteExtensions/FeedExtensions.cs]: Not sure I understand why we need this method that looks identical to previous. Can't the previous one just call this one?

File: Kudu.Core/SiteExtensions/FeedExtensions.cs [hunk 2]
```diff
@@ -245,7 +262,7 @@ public static class FeedExtensions
             }
         }
 
-        private static async Task<T> GetResourceAndValidateAsync<T>(this SourceRepository srcRepo) where T : class, INuGetResource
+        public static async Task<T> GetResourceAndValidateAsync<T>(this SourceRepository srcRepo) where T : class, INuGetResource
```
Comment (davidebbo) [inline Kudu.Core/SiteExtensions/FeedExtensions.cs]: `internal` is preferred here

File: Kudu.Core/SiteExtensions/SiteExtensionManager.cs [hunk 3]
```diff
@@ -87,13 +87,16 @@ public SiteExtensionManager(IContinuousJobsManager continuousJobManager, ITrigge
                             .OrderByDescending(p => p.DownloadCount);
             }
 
-            using (tracer.Step("Convert search result to SiteExtensionInfos"))
+            using (tracer.Step("Convert search result to SiteExtensionInfos with max concurrent requests: {0}", System.Environment.ProcessorCount))
             {
+                // GetResourceAsync is not thread safe
```
Comment (davidebbo) [inline Kudu.Core/SiteExtensions/SiteExtensionManager.cs]: Is the comment referring to `GetResourceAndValidateAsync` or a different method called `GetResourceAsync`?

Comment (watashiSHUN) [inline Kudu.Core/SiteExtensions/SiteExtensionManager.cs]: public static async Task<T> GetResourceAndValidateAsync<T>(this SourceRepository srcRepo) where T : class, INuGetResource
        {
            var resource = await srcRepo.GetResourceAsync<T>();

GetResourceAsync is the nuget method being called in the end

**Pull request closed without merging**